src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -128,25 +134,26 @@ public override void InitializeJobExtension(IExecutionContext executionContext,
                 return;
             }
 
-            // We only support checkout one repository at this time.
-            if (!TrySetPrimaryRepositoryAndProviderInfo(executionContext))
+            // We set the variables based on the 'self' repository
+            if (!TryGetRepositoryInfo(executionContext, "self", out RepositoryInfo repoInfo))
```

Comment:
Does this break for designer builds?

Reply:
I need to update the comment or code here to make this more obvious. But the method will ignore the repo alias passed in if there is only one repo. This can be improved. I will fix it.

Reply:
I fixed the code so we didn't specify 'self' anymore. It isn't necessary in this class since we always want the primary repository.